src/mdesktopentry.cpp

```diff
@@ -243,32 +288,51 @@ QString MDesktopEntry::fileName() const
 
 bool MDesktopEntry::contains(const QString &key) const
 {
+    if (key.startsWith(DesktopEntrySection + "/")) {
+        return d_ptr->desktopEntriesMap.contains(key.mid(DesktopEntrySection.length() + 1));
+    }
     return d_ptr->desktopEntriesMap.contains(key);
 }
 
 bool MDesktopEntry::contains(const QString &group, const QString &key) const
 {
-    return d_ptr->desktopEntriesMap.contains(group + '/' + key);
+    if (group != DesktopEntrySection) {
+        return false;
+    }
+    return d_ptr->desktopEntriesMap.contains(key);
 }
 
 QString MDesktopEntry::value(const QString &key) const
 {
+    if (key.startsWith(DesktopEntrySection + "/")) {
+        return d_ptr->desktopEntriesMap.value(key.mid(DesktopEntrySection.length() + 1));
+    }
     return d_ptr->desktopEntriesMap.value(key);
 }
 
 QString MDesktopEntry::value(const QString &group, const QString &key) const
 {
-    return d_ptr->desktopEntriesMap.value(group + '/' + key);
+    if (group != DesktopEntrySection) {
+        return QString();
```

Comment:
Keep in mind that it is also public API, so you should be quite careful with regards to behaviour changes, as we can't fix everything using it